modules/core/src/main/java/org/apache/ignite/internal/processors/datastreamer/DataStreamerImpl.java

```diff
@@ -1143,6 +1252,8 @@ else if (f.isDone()) {
     /** {@inheritDoc} */
     @SuppressWarnings("ForLoopReplaceableByForEach")
     @Override public void flush() throws CacheException {
+        flushAllThreadsBufs();
```

Comment:
Why this executed before `enterBusy()` call?

Reply:
because enterBusy throws IllegalStateException when there are no data nodes to flush (according to testNoDataNodesOnFlush in DataStreamerImplSelfTest)

Reply:
And why is that?